### ISSUE TITLE
New rule: no-mimic

### ISSUE THREAD

**Issue by cschuller** [eslint/eslint#4745]
**When does this rule warn?**
Warning for Unicode characters substituting common ASCII characters. [See Unicode to create tragedy](https://github.com/reinderien/mimic)

```javascript
var fοo ＝ 'bar';
```

Will throw an _Uncaught SyntaxError: Unexpected token ILLEGAL_ because the Unicode character 0xff1d was used instead of the '=' ASCII character.

```javascript
function foo() { return 'bar'; }
```
Will throw an _Uncaught SyntaxError: Unexpected token ILLEGAL_ because the Unicode character 0x37e was used instead of the semicolon (';') ASCII character.

**Is this rule preventing an error or is it stylistic?**
Will prevent an error. I assume that probability of occurrence is low. 

**Why is this rule a candidate for inclusion instead of creating a custom rule?**
It's a very generic rule that can easily be used and would apply to all kind of JavaScript code.

**Are you willing to create the rule yourself?**
YES

**Comment by michaelficarra** [comment]
You won't be able to create this rule because that would fail to parse. So the linter is already catching these for you.

**Comment by nzakas** [comment]
I think this is best as a custom rule. It's pretty difficult to insert a Unicode character that both looks like an alpha character and is a valid identifier character, so I don't think this is common enough to be a core rule.

**Comment by cschuller** [comment]
Yes you are right. And what about function names and variable declarations?

no-mimic-names

This would then be a stylistic rule.

**Comment by nzakas** [comment]
Those are also identifiers.

**Comment by cschuller** [comment]
Agree with closing this issue and I also agree that's pretty difficult to insert it.

I got the idea from a [sonar](http://www.sonarqube.org/) rule. In our team we moved away from sonar as a linting tool to only use eslint.

By the way, thanks for your great work.
